modern/src/App.js

```diff
@@ -321,6 +321,82 @@ function Layer({ node, id, image, x, y }) {
   );
 }
 
+function AnimatedLayer({
+  node,
+  id,
+  image,
+  x,
+  y,
+  w,
+  h,
+  framewidth,
+  frameheight,
+}) {
+  if (image == null) {
+    console.warn("Got an AnimatedLayer without an image. Rendering null", id);
+    return null;
+  }
+
+  const img = node.js_imageLookup(image.toLowerCase());
+  const frameNum = node.getcurframe();
+
+  const params = {};
+  if (x !== undefined) {
+    params.left = Number(x);
+  }
+  if (y !== undefined) {
+    params.top = Number(y);
+  }
+
+  if (framewidth !== undefined) {
+    params.width = Number(framewidth);
+    params.backgroundPositionX =
+      -(Number(framewidth) * frameNum) % Number(img.w);
+  } else if (w !== undefined) {
+    params.width = Number(w);
+    params.backgroundPositionX = -(Number(w) * frameNum) % Number(img.w);
+  } else {
+    if (img.w !== undefined) {
+      params.width = Number(img.w);
+    }
+    if (img.x !== undefined) {
+      params.backgroundPositionX = -Number(img.x);
+    }
+  }
+
+  if (frameheight !== undefined) {
+    params.height = Number(frameheight);
+    params.backgroundPositionY = -Number(frameheight) * frameNum;
+  } else if (h !== undefined) {
+    params.height = Number(h);
+    params.backgroundPositionY = -Number(h) * frameNum;
+  } else {
+    if (img.h !== undefined) {
+      params.height = Number(img.h);
+    }
+    if (img.y !== undefined) {
+      params.backgroundPositionY = -Number(img.y);
+    }
+  }
```

Comment:
These two blocks are nearly identical (height/width) I wonder if we should abstract something. Might be tricky, but I bet something is possible. Something which takes in the needed values (axis independent) and returns the `{backgroundOffset, size}` or something.

Reply:
Sounds good!